tidy3d/plugins/mode/mode_solver.py

```diff
@@ -10,6 +10,9 @@
 import pydantic.v1 as pydantic
 import xarray as xr
 
+from joblib import Parallel, delayed
```

Comment:
`joblib` will have to be added as a dependency in pyproject.toml right? Curious why you chose to use that instead of the native python `multiprocessing`? As far as I understand `joblib` seems to use that under the hood?

One thing I'm noticing is that you're not explicitly setting the openmp, etc. num threads to 1. This was a problem when you were doing multiprocessing? Did using `joblib` somehow remove the need for that?

Reply:
@momchil-flex I initially tried using multiprocessing but was getting errors like this `AttributeError: Can't pickle local object 'ModeSolver._solve_all_freqs.<locals>.<lambda>'.  Also, was getting errors when the make_simulation function that creates the mode solver loads gds files. Tried messing around with multiprocessing including using libraries like dill for serializing but couldn't make it work.  Multithreading for concurrency works fine but since this task is compute limited we want multiprocessing. Then I found this library and it worked right away and was embarrassingly simple like they say in their website :) By default it is using backend="loky" which sets multiple process workers with n_jobs defining the number of cpu cores it is using. You can also switch to backend= "threading" for multithreading but this is not what we want for processing multiple frequencies in parallel. More details here https://joblib.readthedocs.io/en/latest/generated/joblib.Parallel.html

Reply:
Yeah I could help you go around the "can't pickle object" problem which we've hit multiple times by now (and the answer is not to use dill, even though that's what you find on StackOverflow :) You just need to re-arrange things a bit usually). But I do like this joblib library and we may even switch to that in other places! It seems like it's pretty light and has no dependencies for basic functionality.

Reply:
yeah, would love to learn how you fixed the "can't pickle object" error. Yeah, joblib just looks simple and very readable!